core/src/thrift/tachyon.thrift

```diff
@@ -148,7 +154,7 @@ service MasterService {
    * return the command from master to worker. addedBlockIds maps from id of storage directory
    * to the blocks added in it.
    */
-  Command worker_heartbeat(1: i64 workerId, 2: list<i64> usedBytesOnTiers, 3: list<i64> removedBlockIds,
+  Command worker_heartbeat(1: i64 workerId, 2: list<i64> usedBytesOnTiers, 3: map<i64, list<i64>> removedBlockIds,
```

Comment:
Originally, when the worker evicted a block, it would only report it as adding the block to the new storage id, and the master would update its entry for the worker's storage dir. Now, with multiple storage ids, the worker needs to report which blocks were removed from which storage dirs during eviction, so the master knows exactly what to add and delete.